### ISSUE TITLE
managed counter included in daily switchover like normal counter

### ISSUE THREAD

**Issue by WillemD61** [domoticz/domoticz#5447]
WIKI states: Managed counter will not transfer automatically every day data from device status to history log.

However, this is not true. I tested a managed counter of type counter and this is included in the addcalendarupdatemeter() function like any normal counter. A meter_calendar record is created automatically by the daily switchover routine.

If I insert a meter_calendar entry with updateHistory() like suggested in the documentation, then I get double records in the meter_calendar table.

**Comment by gizmocuz** [comment]
No idea where this is done but I guess here: **CSQLHelper::AddCalendarUpdateMeter()**
So basically, this routine should do nothing for managed meters as all is managed externally correct?

This has been done for quite some time, maybe from the beginning... I have to issues to change this but maybe some people depend on it?

**Comment by WillemD61** [comment]
From the documentation it seems that it was indeed the intention to have everything managed externally.  I don't know how people are using it (I was just testing it to find out whether it would be suitable for negative cost tracking) and I can understand your reluctance to change it.

The alternative could be to adapt the documentation. If you do that, I suggest you also add a clarification that for managed counters the counterToday is always 0 (even if you add data with updateHistory into the meter tabel)

And also a note on graphs:
1) for a normal counter the hourly graph is showing the difference in the value field of one hour to the other (i.e. difference between counter values)
2) for a managed counter, the hourly graph is not built from the value field but just shows whatever is in the usage field in the meter table.

If I find anything else, I will let you know here.

**Comment by Zuikkis** [comment]
@gizmocuz ,

There's some talk about this in the forums:

https://www.domoticz.com/forum/viewtopic.php?p=295887#p295887

Remember the database I sent you, that had double entries in Meter_Calendar? This Domoticz daily switchover is what creates the double entries. 

dzVents updateHistory() call never creates a double entry in database. If there is a matching hour or a matching day, the old database entry is just updated. You could call it every minute and it doesn't matter, only the latest data will be in database.

However at midnight Domoticz switchover creates a double entry, usually with invalid data.

Only way you don't get double entries if you call updateHistory() only once and AFTER midnight. Then updateHistory() will overwrite the invalid entry created by the daily switchover.

If it's possible to remove managed counters from the daily switchover, please do.. :) I don't think anyone can be relying on this faulty behaviour.

**Comment by gizmocuz** [comment]
@WillemD61 ,


I noticed something in the code. Do you have anything in the Options field for this device in the DeviceStatus table?

There seems to be an option named 'DisableLogAutoUpdate' , when this is 'true ' those meter/multi_meter tables won't be touched by the daily routines

See 'Note on counters' in

https://www.domoticz.com/wiki/Domoticz_API/JSON_URL%27s

Could you tell me if this solves your problems?

I think this needs to be default for managed counters, but if this solves the issue I know where to add a check if it is a managed counter and do nothing

**Comment by WillemD61** [comment]
I created a new managed counter, there is nothing in the options fields for most types of managed counter. Only if you choose counter type and fill in "value quantity"and "value unit" then there is : "ValueQuantity:Y29zdHM=;ValueUnits:RVVSTw==" so the filled in fields get encoded.

At the link you provide, it states you need to set that DisableLogAutoUpdate option yourself, for example with a python program. I guess I can therefore add it to the end with a semicolon in between? 

I will certainly test. I think still the problem remains the "updateHistory()" does not process negative numbers. I will redo my tests.

By the way, the daily graph for KWh counters can handle ups and downs no problem. I don't understand why a counter cannot.

**Comment by Zuikkis** [comment]
BTW, here is the issue where min/max is fixed to first/last for the kWh counter:
https://github.com/domoticz/domoticz/issues/4380

I guess you could apply similar patch to other counters as well.

**Comment by gizmocuz** [comment]
Yes, but the wiki also mentions you can set this option yourself bij a json call

So please test if this solves the problem, and if it does, I will add a check on all the positions where this is used for managed counters

**Comment by WillemD61** [comment]
I tried with the following options field in devicestatus (ValueQuantity and ValueUnits were set by editing the device using Domoticz screens, the DisableLogAutoUpdate was added using sqlite3):

ValueQuantity:Q29zdHM=;ValueUnits:RVVSTw==;DisableLogAutoUpdate:dHJIZQ==

and the device is still included in the daily switchover.

But I don't mind if this is not solved/changed, since I am not going to use counters for cost tracking because of the multitude of issues when costs decrease.

**Comment by WillemD61** [comment]
Oops, I made an error, should have used dHJ1ZQ== instead of dHJIZQ==

Will test again

**Comment by WillemD61** [comment]
I confirm that adding "DisableLogAutoUpdate:dHJ1ZQ==" to the options field in the devicestatus table excludes the device from the daily switchover at midnight. So, as result, no record is created in the meter_calendar table for the device.

For managed counters the updateHistory() call can then be used to add a entry to the meter_calendar table. For other counters a JSON call needs to be used, but for those normal counters first also "AddDBLogEntry:dHJ1ZQ==" needs to be set. (if it is not set then only devicestatus is updated by the JSON call)

Issue can be closed, but I suggest to clarify the documentation (was already done yesterday for AddDBLogEntry).

Just to add: any double records are not created by either the updateHistory or the JSON call. These calls just replace any existing record. It is the midnight switchover routine that adds a second record even if one was created by the updateHistory or JSON earlier (unless the DisableLogAutoUpdate is set of course).
